ibis/pyspark/compiler.py

```diff
@@ -1011,3 +1058,296 @@ def compile_cumulative_max(t, expr, scope, *, window, **kwargs):
     return _handle_cumulative_operation(
         t, expr, scope, F.max, window=window, **kwargs
     )
+
+
+# -------------------------- Temporal Operations ----------------------------
+
+# Ibis value to PySpark value
+_time_unit_mapping = {
+    'Y': 'year',
+    'Q': 'quarter',
+    'M': 'month',
+    'W': 'week',
+    'D': 'day',
+    'h': 'hour',
+    'm': 'minute',
+    's': 'second'
+}
+
+
+@compiles(ops.Date)
+def compile_date(t, expr, scope, **kwargs):
+    op = expr.op()
+    src_column = t.translate(op.arg, scope)
+    return F.to_date(src_column).cast('timestamp')
```

Comment:
why cast date to timestamp?

Reply:
Casting to timestamp will result in the correct `datetime64[ns]` when materializing the data with `toPandas()`. Otherwise, the incorrect `object` data type is returned.